imap/sync_support.c

```diff
@@ -2011,6 +2013,13 @@ static int sync_prepare_dlists(struct mailbox *mailbox,
     encode_annotations(kl, mailbox, NULL, annots);
     sync_annot_list_free(&annots);
 
+    /* always send mailbox userflags */
+    ufl = dlist_newlist(kl, "USERFLAGS");
+    for (userflag = 0; userflag < MAX_USER_FLAGS; userflag++) {
+        if (mailbox->h.flagname[userflag])
+            dlist_setatom(ufl, "USERFLAG", mailbox->h.flagname[userflag]);
```

Comment:
Should use `dlist_set_flag` here maybe - it allows things which are legal flags but not legal atoms (yeah, really) to go on the wire naturally.

Reply:
Oh, I missed that `dlist_set_flag` exists.  Yeah I'll use that!